dipy/workflows/reconst.py

```diff
@@ -161,13 +161,101 @@ def run(self, input_files, bvalues, bvectors, mask_files, b0_threshold=0.0,
             logging.info('DTI metrics saved in {0}'.
                          format(os.path.dirname(oevals)))
 
+    def get_tensor_model(self, gtab):
+        return TensorModel(gtab, fit_method="WLS")
+
+    def get_fitted_tensor(self, data, mask, bval, bvec, b0_threshold=0):
+
+        logging.info('Tensor estimation...')
+        bvals, bvecs = read_bvals_bvecs(bval, bvec)
+        gtab = gradient_table(bvals, bvecs, b0_threshold=b0_threshold)
+
+        tenmodel = self.get_tensor_model(gtab)
+        tenfit = tenmodel.fit(data, mask)
+
+        return tenfit, gtab
+
+
+class ReconstDtiRestoreFlow(ReconstDtiFlow):
+    @classmethod
+    def get_short_name(cls):
+        return 'dti_restore'
+
+    def run(self, input_files, bvalues, bvectors, mask_files, b0_threshold=0.0,
+            save_metrics=[], sigma=None, jacobian=True,
+            out_dir='', out_tensor='tensors.nii.gz', out_fa='fa.nii.gz',
+            out_ga='ga.nii.gz', out_rgb='rgb.nii.gz', out_md='md.nii.gz',
+            out_ad='ad.nii.gz', out_rd='rd.nii.gz', out_mode='mode.nii.gz',
+            out_evec='evecs.nii.gz', out_eval='evals.nii.gz'):
+
+        """ Workflow for tensor reconstruction and for computing DTI metrics.
+            Performs a tensor reconstruction on the files by 'globing'
+            ``input_files`` and saves the DTI metrics in a directory specified by
+            ``out_dir``.
 
-def get_fitted_tensor(data, mask, bval, bvec, b0_threshold=0):
-    logging.info('Tensor estimation...')
-    bvals, bvecs = read_bvals_bvecs(bval, bvec)
-    gtab = gradient_table(bvals, bvecs, b0_threshold=b0_threshold)
+            Parameters
+            ----------
+            input_files : string
+                Path to the input volumes. This path may contain wildcards to
+                process multiple inputs at once.
+            bvalues : string
+                Path to the bvalues files. This path may contain wildcards to use
+                multiple bvalues files at once.
+            bvectors : string
+                Path to the bvalues files. This path may contain wildcards to use
+                multiple bvalues files at once.
+            mask_files : string
+                Path to the input masks. This path may contain wildcards to use
+                multiple masks at once. (default: No mask used)
+            b0_threshold : float, optional
+                Threshold used to find b=0 directions (default 0.0)
+            save_metrics : variable string, optional
+                List of metrics to save.
+                Possible values: fa, ga, rgb, md, ad, rd, mode, tensor, evec, eval
+                (default [] (all))
+            sigma : float, optional
+                An estimate of the variance. (default None)
```

Comment:
None was the default in restore_fit_tensor but it actually raises an exception. I moved sigma out of optionals and is now requiered. 

For weighting, from what I can see, it is only passable to NLLS and not Restore.